docs/source/code-of-conduct.rst

```diff
@@ -0,0 +1,274 @@
+   already know it. Ask before you teach! Don't assume what people's
+   knowledge gaps are.
+-  `Feigning or exaggerating
+   surprise <https://www.recurse.com/manual#no-feigned-surprise>`__ when
+   someone admits to not knowing something.
+-  "`Well-actuallies <https://www.recurse.com/manual#no-well-actuallys>`__"
+-  Other conduct which could reasonably be considered inappropriate in a
+   professional or community setting.
+
+Scope
+-----
+
+This Code of Conduct applies both within spaces involving this project
+and in other spaces involving community members. This includes the
+repository, its Pull Requests and Issue tracker, its Twitter community,
+private email communications in the context of the project, and any
+events where members of the project are participating, as well as
+adjacent communities and venues affecting the project's members.
+
+Depending on the violation, the maintainers may decide that violations
+of this code of conduct that have happened outside of the scope of the
+community may deem an individual unwelcome, and take appropriate action
+to maintain the comfort and safety of its members.
+
+.. _coc-other-community-standards:
+
+Other Community Standards
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As a project on GitHub, this project is additionally covered by the
+`GitHub Community
+Guidelines <https://help.github.com/articles/github-community-guidelines/>`__.
+
+Enforcement of those guidelines after violations overlapping with the
+above are the responsibility of the entities, and enforcement may happen
+in any or all of the services/communities.
+
+Maintainer Enforcement Process
+------------------------------
+
+Once the maintainers get involved, they will follow a documented series
+of steps and do their best to preserve the well-being of project
+members. This section covers actual concrete steps.
+
+
+.. _coc-contacting-maintainers:
+
+Contacting Maintainers
+~~~~~~~~~~~~~~~~~~~~~~
+
+As a small and young project, we don't yet have a Code of Conduct
+enforcement team. Hopefully that will be addressed as we grow, but for
+now, any issues should be addressed to `Nathaniel J. Smith
+<https://github.com/njsmith>`__, via `email <mailto:njs@pobox.com>`__
+or any other medium that you feel comfortable with.
+
+
+.. _coc-further-enforcement:
+
+Further Enforcement
+~~~~~~~~~~~~~~~~~~~
+
+If you've already followed the :ref:`initial enforcement steps
+<coc-when-something-happens>`, these are the steps maintainers will
+take for further enforcement, as needed:
+
+1. Repeat the request to stop.
+2. If the person doubles down, they will have offending messages removed
+   or edited by a maintainers given an official warning. The PR or Issue
+   may be locked.
+3. If the behavior continues or is repeated later, the person will be
+   blocked from participating for 24 hours.
+4. If the behavior continues or is repeated after the temporary block, a
+   long-term (6-12mo) ban will be used.
```

Comment:
I notice there's no contingency for a permanent ban. Is that on purpose?

Reply:
Oh, hmm, interesting. 99% of the text here is straight from the WeAllJS code of conduct (via [weallbehave](https://github.com/WeAllJS/weallbehave)); I hadn't noticed this particular detail. Looking at it in context though, I think I'm OK with it – one of the things I like about this CoC is that it has clear and straightforward guidance for "regular" grumpiness and jerkitude, while also being clear that if someone's trying to game the system or acting totally beyond the pale then That Will Be Dealt With. This part is in the former section. Good catch though, and if you have more thoughts I'd be interested to hear, I'm not super confident about all this.

Reply:
At the very least I would have a step 5 say that if the behaviour continues then the ban will be permanent. Considering at that point they have misbehaved 3 separate times over the span of at least six months that's enough time to learn how to behave online appropriately. And later on there is a mention that steps may be skipped as deemed necessary, so that covers egregious behaviour that warrants skipping the long-term ban and going straight to a permanent one.

Reply:
Fair enough! Done.